### PR TITLE
Fix issue with zio.test.Gen.fromIterable generating deterministic output

### PR DESCRIPTION
**Pull request by melmathari**
Adjust Gen composition to ensure that combining Gen.fromIterable with other generators in different orders results in non-deterministic output. This resolves issue #9101.

/claim #9101

The key changes:

- Modified fromIterable to include a random element by using ZStream.fromZIO(Random.nextInt) before generating the values
- This ensures that when combined with other random generators like Gen.uuid, the randomization behavior remains consistent regardless of order

### CODE REVIEW

**Comment by melmathari** [comment]
@jdegoes @felix-hedenstrom Could you please review this pull request when you have a moment? Thank you for your contributions!

**Comment by nocodeventure-nl** [comment]
@felix-hedenstrom Do you have a moment to look at this? Thank you!

**Comment by soujiro32167** [comment]
Could this also be applied to `Gen.concatAll`? It would help resolve https://github.com/zio/zio/issues/9300

**Comment by melmathari** [comment]
> Could this also be applied to `Gen.concatAll`? It would help resolve #9300

Change added

**Comment by melmathari** [comment]
@kyri-petrou  Can you please review this PR at your earliest convience? Thanks!

Also let me know if I should remove the last change, to accomodate for #9300

**Comment by melmathari** [comment]
Test failing unrelated to our changes....

- TextCodecPipelineSpec - Text Decoders - utfDecode - UTF-16LE with BOM
        Exception in thread "zio-fiber-" java.nio.charset.MalformedInputException: Input length = 2
        	at zio.stream.ZPipeline$.$anonfun$decodeCharsWithDecoder$3(ZPipeline.scala:996)
        	at zio.ZIO$.$anonfun$fail$1(ZIO.scala:3237)
        	at zio.ZIO$.$anonfun$failCause$3(ZIO.scala:3246)
        	at zio.FiberRef$unsafe$$anon$2.$anonfun$getWith$1(FiberRef.scala:474)

File: test/shared/src/main/scala/zio/test/Gen.scala [hunk 1]
```diff
@@ -436,14 +451,43 @@ object Gen extends GenZIO with FunctionVariants with TimeVariants {
...
+    if (as.isEmpty) empty
+    else
+      Gen {
+        // Check if the iterable is infinite by trying to convert to Vector
+        val asVec =
+          try {
+            Some(as.toVector)
+          } catch {
+            case _: OutOfMemoryError => None
+          }
```
Comment (kyri-petrou) [inline test/shared/src/main/scala/zio/test/Gen.scala:471]: I'm sorry but we really can't do this. Besides making things extremely slow as we'll need to generate collections of several GBs in size, catching an OOM Error should never _ever_ be used as an expected condition

**Review by kyri-petrou — CHANGES_REQUESTED** [review]
Thank you for the contribution! Unfortunately we really can't go with an implementation that relies on causing an OOM error to check whether a generator is finite. I'm going to leave this PR open for now in case you have any other ideas on how to check for infinite generators, but that is something that absolutely needs to be addressed

**Pull request closed without merging**